src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -1105,27 +1106,37 @@ sealed class RsDiagnostic(
             )
     }
 
-    class MissingFieldsInTuplePattern(pat: RsPat, private val declaration: RsFieldsOwner) : RsDiagnostic(pat) {
+    class MissingFieldsInTuplePattern(
+        pat: RsPat,
+        private val declaration: RsFieldsOwner,
+        private val expectedAmount: Int,
+        private val actualAmount: Int
+    ) : RsDiagnostic(pat) {
         override fun prepare(): PreparedAnnotation {
             val itemType = if (declaration is RsEnumVariant) "Enum variant" else "Tuple struct"
+            val fieldOrFields = if (expectedAmount == 1) "field" else "fields"
```

Comment:
`com.intellij.openapi.util.text.StringUtil#pluralize(java.lang.String)` can be usefull here

Reply:
And also I think it's worth to inline this expression to avoid weird variable name